CONTRIBUTING.rst

```diff
@@ -0,0 +1,678 @@
+  flake8                              Runs flake8.
+  forbid-tabs                         Fails if tabs are used in the project.
+  insert-license                      Adds licences for most file types.
+  lint-dockerfile                     Lints a dockerfile.
+  mixed-line-ending                   Detects if mixed line ending is used (\r vs. \r\n).
+  mypy                                Runs mypy.
+  pylint                              Runs pylint.
+  shellcheck                          Checks shell files with shellcheck.
+  yamllint                            Checks yaml files with yamllint.
+
+
+Using Pre-commit Hooks
+----------------------
+
+After installation, pre-commit hooks are run automatically when you commit the
+code. But you can run pre-commit hooks manually as needed.
+
+-   Run all checks on your staged files by using:
+
+.. code-block:: bash
+
+    pre-commit run
+
+
+-   Run only mypy check on your staged files by using:
+
+.. code-block:: bash
+
+    pre-commit run mypy
+
+
+-   Run only mypy checks on all files by using:
+
+.. code-block:: bash
+
+    pre-commit run mypy --all-files
+
+
+-   Run all checks on all files by using:
+
+.. code-block:: bash
+
+    pre-commit run --all-files
+
+
+-   Skip one or more of the checks by specifying a comma-separated list of
+    checks to skip in the SKIP variable:
+
+.. code-block:: bash
+
+    SKIP=pylint,mypy pre-commit run --all-files
+
+
+You can always skip running the tests by providing ``--no-verify`` flag to the
+``git commit`` command.
+
+To check other usage types of the pre-commit framework, see `Pre-commit website <https://pre-commit.com/>`__.
+
+Travis CI Testing Framework
+===========================
+
+Airflow test suite is based on Travis CI framework as running all of the tests
+locally requires significant setup. You can set up Travis CI in your fork of
+Airflow by following the `Travis
+CI Getting Started
+guide <https://docs.travis-ci.com/user/getting-started/>`__..
+
+
+There are two different options available for running Travis CI, and they are
+set up on GitHub as separate components:
+
+-   **Travis CI GitHub App** (new version)
+-   **Travis CI GitHub Services** (legacy version)
+
+Using Travis CI GitHub App (new version)
+----------------------------------------
+
+-   Once `installed <https://github.com/apps/travis-ci/installations/new/permissions?target_id=47426163>`__,
+    configure the Travis CI GitHub App at
+    `Configure Travis CI <https://github.com/settings/installations>`__.
+
+-   Set repository access to either "All repositories" for convenience, or "Only
+    select repositories" and choose ``USERNAME/airflow`` in the drop-down menu.
+
+-   Access Travis CI for your fork at `<https://travis-ci.com/USERNAME/airflow>`__.
+
+Using Travis CI GitHub Services (legacy version)
+------------------------------------------------
+
+..
+
+    The apache/airflow project is still using the legacy version.
```

Comment:
Elena:  //in addition to the new one?
Jarek: No: So far airflow uses the legacy version only

Reply:
why then are you talking about the new one?

Reply:
Any new fork/setup should use the new version. BTW. We are soon going to get rid of Travis hopefully for apache project so this whole legacy/new split might be removed.